test/Microsoft.Extensions.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -36,6 +36,18 @@ public void GetRequiredService_Throws_WhenNoServiceRegistered()
         }
 
         [Fact]
+        public void IRequiredServiceAwareService_GetRequiredService_Throws_WhenNoServiceRegistered()
+        {
+            // Arrange
+            var serviceProvider = new RequiredServiceAwareServiceProvider();
+
+            // Act + Assert
+            ExceptionAssert.Throws<InvalidOperationException>(
+                () => ((IServiceProvider)serviceProvider).GetRequiredService<IFoo>(),
```

Comment:
@smbecker @pranavkm How does this look like? (yeah I could avoid `var` here and avoided the cast, but we usually are using `var` everywhere..)